Tasks/JavaToolInstaller/javatoolinstaller.ts

```diff
@@ -0,0 +1,108 @@
+        fs.readdirSync(extractLocation)
+        .forEach((item: string) => {
+            let itemPath = path.join(extractLocation, item);
+            taskLib.rmRF(itemPath);
+        });
+    }
+
+    if (version) { //This version of Java JDK is already in the cache. Use it instead of downloading again.
+        console.log(taskLib.loc("Info_ResolvedToolFromCache", version));
+    }
+    else if (fromAzure) { //Download JDK from an Azure blob storage location and extract.
+        try {
+            compressedFileExtension = getFileEnding(taskLib.getInput('azureCommonVirtualPath', true));
+        
+            let azureDownloader = new azureStorageDownloader.AzureStorageArtifactDownloader(taskLib.getInput('azureResourceManagerEndpoint', true), taskLib.getInput('azureStorageAccountName', true), taskLib.getInput('azureContainerName', true), taskLib.getInput('azureCommonVirtualPath', false));
+            await azureDownloader.downloadArtifacts(extractLocation, '*' + compressedFileExtension);
+            await sleepFor(250); //Wait for the file to be released before extracting it.
+
+            let extractSource = buildFilePath(extractLocation, compressedFileExtension);
+            jdkDirectory = new JavaFilesExtractor().unzipJavaDownload(extractSource, compressedFileExtension, extractLocation);
+        } catch (err) {
+            taskLib.error(err.message);
+            taskLib.setResult(taskLib.TaskResult.Failed, err.message);
+        }
+    }
+    else { //JDK is in a local directory. Extract to specified target directory.
+        try{
+            compressedFileExtension = getFileEnding(taskLib.getInput('jdkPath', true));
+            jdkDirectory = new JavaFilesExtractor().unzipJavaDownload(taskLib.getInput('jdkPath', true), compressedFileExtension, extractLocation);
+        } catch (err) {
+            taskLib.error(err.message);
+            taskLib.setResult(taskLib.TaskResult.Failed, err.message);
+        }
+    }
+
+    let extendedJavaHome = 'JAVA_HOME_' + versionSpec + '_' + taskLib.getInput('jdkArch', true);
+    console.log(taskLib.loc("SetJavaHome", jdkDirectory));
+    taskLib.setVariable('JAVA_HOME', jdkDirectory);
+    taskLib.setVariable(extendedJavaHome, jdkDirectory);
+} 
+
+function sleepFor(sleepDurationInMillisecondsSeconds): Promise<any> {
+    return new Promise((resolve, reeject) => {
+        setTimeout(resolve, sleepDurationInMillisecondsSeconds);
+    });
+}
+
+function buildFilePath(localPathRoot: string, fileEnding: string): string {
+    const azureFileSource: string = taskLib.getInput('azureCommonVirtualPath', true);
+    var fileName = azureFileSource.split(/[\\\/]/).pop();
+    var extractSource = path.join(localPathRoot, fileName);
+    console.debug("Extracting JDK from: "+ extractSource);
+
+    return extractSource;
+}
+
+function getFileEnding(file: string): string {
+    var fileEnding = "";
+
+    if(file.endsWith(".tar")) {
```

Comment:
space after "if" - I just happen to see this. I make this mistake all the time. You may want to turn on linting in VSCode to catch these problems.